ale/drivers/selene_drivers.py

```diff
@@ -520,33 +530,302 @@ def detector_start_sample(self):
 
     @property
     def detector_start_line(self):
-        if self.spacecraft_direction < 0:
-            return super().detector_start_line
-        elif self.spacecraft_direction > 0:
-            return 1
+        return 1
+
+    @property
+    def sensor_model_version(self):
+        """
+        Returns
+        -------
+        : int
+          ISIS sensor model version
+        """
+        return 2
+
+class KaguyaTcIsisLabelNaifSpiceDriver(LineScanner, IsisLabel, NaifSpice, Driver):
+    """
+    """
+    @property
+    def instrument_id(self):
+        """
+        Id takes the form of LISM_<INSTRUMENT_ID>_<SD><COMPRESS><SWATH> where
+
+        INSTRUMENT_ID = TC1/TC2
+        SD = S/D short for single or double, which in turn means whether the
+        label belongs to a mono or stereo image.
+
+        COMPRESS = D/T short for DCT or through, we assume image has been
+        decompressed already
+
+        SWATCH = swatch mode, different swatch modes have different FOVs
+
+        Returns
+        -------
+        : str
+          instrument id
+        """
+        instrument = super().instrument_id
+        swath = self.label["IsisCube"]["Instrument"]["SwathModeId"][0]
+        sd = self.label["IsisCube"]["Archive"]["ProductSetId"].split("_")[1].upper()
+
+        inst_id = "LISM_{}_{}T{}".format(instrument, sd, swath)
+        return inst_id
 
     @property
-    def spacecraft_direction(self):
+    def sensor_frame_id(self):
         """
-        Gets the moving direction of the spacecraft from the label, where -1 is moving
-        as intended and 1 is moving inverted.
+        Returns the sensor frame id.  Depends on the instrument that was used to
+        capture the image.
 
         Returns
         -------
         : int
-          Moving direction of the spacecraft
+          Sensor frame id
         """
-        return int(self.label['SATELLITE_MOVING_DIRECTION'])
+        return spice.namfrm("LISM_{}_HEAD".format(super().instrument_id))
 
+    @property
+    def ikid(self):
+        """
+        Read the ikid from the cube label
+        """
+        if not hasattr(self, "_ikid"):
+            self._ikid = spice.bods2c("LISM_{}".format(super().instrument_id))
+        return self._ikid
 
     @property
-    def sensor_model_version(self):
+    def platform_name(self):
+        """
+        Returns the name of the platform containing the sensor. This is usually
+        the spacecraft name.
+
+        Returns
+        -------
+        : str
+          Name of the platform which the sensor is mounted on
+        """
+        return self.label['IsisCube']['Instrument']['MissionName']
```

Comment:
Shouldn't one of these be spacecraft name and the other mission name? I remember these labels being a little weird in that regard.

Reply:
I will look into this

Reply:
I was able to remove the `spacecraft_name` override here so that resolves this